sdk/identity/Azure.Identity/src/KubernetesProxyHttpHandler.cs

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    /// HTTP message handler that redirects all requests to a Kubernetes token proxy.
+    /// Handles custom CA certificates and monitors CA file changes for automatic reloading.
+    /// </summary>
+    internal class KubernetesProxyHttpHandler : DelegatingHandler
+    {
+        private readonly KubernetesProxyConfig _config;
+        private HttpClientHandler _innerHandler;
+        private byte[] _lastCaData;
+        private readonly object _lock = new object();
+
+        public KubernetesProxyHttpHandler(KubernetesProxyConfig config)
+        {
+            _config = config ?? throw new ArgumentNullException(nameof(config));
+
+            // Create the initial handler
+            _innerHandler = CreateHandler();
+            InnerHandler = _innerHandler;
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            // Check if we need to reload the CA file and recreate the handler
+            if (ShouldReloadHandler())
+            {
+                lock (_lock)
+                {
+                    if (ShouldReloadHandler())
+                    {
+                        var oldHandler = _innerHandler;
+                        _innerHandler = CreateHandler();
+                        InnerHandler = _innerHandler;
+
+                        // Dispose old handler (but not immediately, allow in-flight requests to complete)
+                        Task.Run(() =>
+                        {
+                            Thread.Sleep(1000); // Give in-flight requests time to complete
```

Comment:
Perhaps we can keep a `SemaphoreSlim` to keep track of outstanding requests and do this dispose more deterministically.